framer/Components/SliderComponent.coffee

```diff
@@ -119,17 +143,33 @@ class exports.SliderComponent extends Layer
 			width: @width + @knob.width
 			height: @height + @knob.height
 
+		if @bounded
+			@knob.draggable.constraints =
+				x: 0
+				y: 0
+				width: @width
+				height: @height
+
 		if @width > @height
 			@fill.height = @height
 			@knob.centerY()
+			@knobOverlay.centerY()
 		else
 			@fill.width = @width
 			@knob.centerX()
+			@knobOverlay.centerX()
+
+		@sliderOverlay.center()
 
 	_setRadius: =>
 		radius = @borderRadius
 		@fill.style.borderRadius = "#{radius}px 0 0 #{radius}px"
 
+	_setOverlayRadius: =>
+		@knobOverlay.borderRadius = @knob.borderRadius
+
+	@define "bounded", @simpleProperty("bounded", false)
```

Comment:
@benjaminnathan @koenbok Is `bounded` the best term here to express that the outer edges of the knob will be constraint to the slider track (instead of the knob's center point). Alternatives could be:

- constrainToTrack (and move the property onto `knob`)
- constrainKnobToTrack
- centerKnob (`true` by default)
- ?